models/data_maps.go

```diff
@@ -0,0 +1,93 @@
+package models
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"math"
+	"time"
+
+	"github.com/gobuffalo/pop"
+	"github.com/gobuffalo/validate"
+	"github.com/gobuffalo/validate/validators"
+)
+
+const fileBytesChunkSize = float64(2817)
+
+type DataMap struct {
+	ID          int       `json:"id" db:"id"`
+	CreatedAt   time.Time `json:"created_at" db:"created_at"`
+	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
+	Status      int       `json:"status" db:"status"`
+	HooknodeIp  string    `json:"hooknode_ip" db:"hooknode_ip"`
+	Message     string    `json:"message" db:"message"`
+	TrunkTx     string    `json:"trunk_tx" db:"trunk_tx"`
+	BranchTx    string    `json:"branch_tx" db:"branch_tx"`
+	GenesisHash string    `json:"genesis_hash" db:"genesis_hash"`
+	ChunkIdx    int       `json:"chunk_idx" db:"chunk_idx"`
+	Hash        string    `json:"hash" db:"hash"`
+	Address     string    `json:"address" db:"address"`
+}
+
+// String is not required by pop and may be deleted
+func (d DataMap) String() string {
+	jd, _ := json.Marshal(d)
+	return string(jd)
+}
+
+// DataMaps is not required by pop and may be deleted
+type DataMaps []DataMap
+
+// String is not required by pop and may be deleted
+func (d DataMaps) String() string {
```

Comment:
Do we need both this function as well as the one that is on line 33?

Reply:
Notice how the one on `33` is singular and this one is plural. Since golang refuses to support generics, we have hacks like this :(

Reply:
Ah, I see.